src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -1145,6 +1175,23 @@ private MethodSymbol PerformForEachPatternOverloadResolution(TypeSymbol patternT
                     // We don't want to allow this case.
                     result = null;
                 }
+                else
+                {
+                    var argsToParams = overloadResolutionResult.ValidResult.Result.ArgsToParamsOpt;
+                    var expanded = overloadResolutionResult.ValidResult.Result.Kind == MemberResolutionKind.ApplicableInExpandedForm;
+                    BindDefaultArguments(
```

Comment:
Did we already have tests for MoveNextAsync/DisposeAsync instance or extension methods with default parameter values?

Reply:
Neither can be extensions. Looks like we have a test for `MoveNextAsync`: `CodeGenAsyncForEachTests.TestMoveNextAsync_WithOptionalParameter`, but not for dispose. I'll add one.

---
In reply to: [549493544](https://github.com/dotnet/roslyn/pull/50105#discussion_r549493544) [](ancestors = 549493544)